### ISSUE TITLE
Research: Best way to show the events of the year on the event's page

### ISSUE THREAD

**Issue by y4izus** [Qiskit/qiskit.org#2943]
There is a need on the events page to let the users know all the events that are gonna take place during the current year. This issue is intended to do a research to find the best/faster approach for reach the goal. 

We already did a first brainstorming session and these are the main ideas we mentioned:
- Look for a JS library that allow us to create a calendar in a fast way. The info can be pulled from the airtable
- Airtable allows us to use an iframe with the calendar view. We decided to decline this option because we cannot style with our CSS the iframe
- Check if with the enterprise airtable account we are able to style the calendar view to make it look similar to qiskit.org
- Look for alternatives. What can we use instead of a calendar?

Remember, the main goal is to show the events on the current year in a friendly way

---

Part of #2696

**Comment by techtolentino** [comment]
Team, I am providing my summary of research and testing, for various solutions to show events for the full year. Please let us know which of these solutions, if any, seem like the best approach to move forward with.



## The "No Date Range Filter" Approach
In this implementation, we simply remove the date range filters from the current list of Upcoming events. As a result, we'll see event cards for all future events on the calendar.

**Pros**
- the size of effort (XS)
- users can see all future events
- responsive UI

**Cons**
- the list is lengthy
- the list is repetitive
- not much visual distinction

**Screenshot**

https://user-images.githubusercontent.com/6276074/217007525-a5fbabac-6c80-46aa-9cc9-efb20415575c.mov





## The "Month by Month Sections" Approach
A slight improvement to removing the date range filter would be to add monthly sections to the page, with anchor tags that allow users to jump to the months they care about.

**Pros**
- the size of effort (SM)
- users can see all future events
- users can navigate to the month of their choice
- responsive UI

**Cons**
- the list is still lengthy
- not much visual distinction

**Screenshot**


https://user-images.githubusercontent.com/6276074/217007720-2c2cacc1-0d84-48af-8896-315afb51ba05.mov




## Using the [Daypilot](https://doc.daypilot.org/calendar/vue-js/) Library
One of the better open-source calendar libraries is Daypilot, which has some support for Vue applications.

**Pros**
- the size of effort (MD - L)
- users can view all events
- users can navigate to the month of their choice
- provides a visual calendar
- can have _some_ customization

**Cons**
- documentation is a little sparse
- calendar will need CSS overrides to match Qiskit branding
- need to code/style your own navigation buttons
- customization can increase the scope
- took some time to even get a prototype working
- works best on desktop and tablet
- current implementation is in Vue 2, which we will need to maintain as we migrate to Nuxt3

**Screenshot**


https://user-images.githubusercontent.com/6276074/217007880-47b861f2-b2d5-4920-9f92-50f907fd49d0.mov




## Using the [FullCalendar.io w/ Vue](https://fullcalendar.io/docs/vue) Library
Another open-source calendar option is FullCalendar.io, which has better documentation.

**Pros**
- the size of effort (MD - L)
- users can view all events
- users can navigate to the month of their choice
- dev can simply provide an .ICS file in the code base
- provides a visual calendar
- can have _some_ customization

**Cons**
- calendar will need CSS overrides to match Qiskit branding
- customization can increase the scope
- works best on desktop and tablet
- current implementation is in Vue 2, which we will need to maintain as we migrate to Nuxt3

**Screenshot**


https://user-images.githubusercontent.com/6276074/217007946-d0d7e539-6bf5-453c-9e52-e652052741cd.mov




## Using the Airtable Calendar iFrame
Using the AirTable embedded code snippet to add an iFrame to the Events page

**Pros**
- the size of effort (XS)
- users can view all events
- users can navigate to the month of their choice
- provides a visual calendar
- works on desktop, tablet, and mobile (allows overflow scrolling)
- no maintenance on the dev part

**Cons**
- cannot customize iFrame from our code base
- (_may be_ able to customize _some_ from within Airtable side?, not sure)
  
  **Screenshot**
  

https://user-images.githubusercontent.com/6276074/217008012-cbfccf4f-41ab-4620-8c91-9b0d9cc5c4f9.mov

**Comment by JRussellHuffman** [comment]
Really nice thorough exploration. 
Do you have one that you prefer/recommend?

My personal preference would be one of the top two options with some subtle tweaks. The calendar options are nice, but they feel a little over-engineered, and I think the first two options are both viable.

For the first option ("The "No Date Range Filter" Approach")
Could there be a "show more" option after a certain number of cards? For example, we only show 20 cards and then a "show more" to reveal the remaining options. This would make it so the "request an event" isn't quite as buried, and the repetition of the later cards is hidden by default, but the user is still able to get to all events for the year. Side note, do we know how frequently the "request an event" is used?

For the second option ("The "Month by Month Sections" Approach")
Would it be possible to make the months a section similar to "Locations" and "Types", and then make that section fixed when scrolling? That way it fits within the UI and the user can easily jump between sections.

**Comment by eddybrando** [comment]
Awesome work @techtolentino!

I agree with @JRussellHuffman and I'm also a fan of the first options.

I like the "show more" button to avoid showing the "request an event" after a long scrolling journey.

Another approach could be to add pagination and only display the events in groups of 12 for instance.

**Comment by anamariarojas123** [comment]
Thanks for all your work on this! 

My preferred is the "month by month" sections and as Russell mentioned, if there's a way to make the months a section similar to locations and types that would be great, or if you could format it so that only the current month shows on the page and then you can change the page via the anchor tags (although I guess they wouldn't be anchor tags at that point). 

Basically, month by month is amazing, and preferred if we could do one month view at a time.

**Comment by techtolentino** [comment]
Some quick notes:

I think the embedded Airtable is the _least engineered_ — we'd simply add a line of code into a component and the implementation would be about 95% done.

Benefits of using the embedded Airtable:
- content would be managed on the Airtable side and the Events team would have control
- dev wouldn't have to manipulate data
- we could simplify a lot by removing the `Upcoming | Past` event tabs
- users know how to use a calendar app, and Airtable is among the best-in-class calendar solutions out there

Disadvantages
- mobile _could_ be better, though we should test if this would be an issue
- not 100% aligned w/ our branding

__

My humble opinion, based on having explored these options... 

**If the calendar had to launch by end of this sprint:** Airtable 

**If the team had another sprint to work on it:** I'd show all events for the upcoming year, allow month filters... and I'd remove the upcoming/past events (unless data proved that users need to know 2022 events)

**Comment by techtolentino** [comment]
Wanted to share a prototype of what the Month filters interaction might look like if they were added to the left side filter sets...

I still feel like this is overengineered. From an experience standpoint, the filter options run off the height of the page (desktop), and the page is a little jumpy, depending on where the user has scrolled to, in combination of what the results could be.



https://user-images.githubusercontent.com/6276074/217952451-f5b33f68-5d63-454a-bcb1-20a157bb119f.mov

**Comment by JRussellHuffman** [comment]
After having done this prototype, are you still most in favor of the airtable embedded calendar? If the airtable calendar is on a new tab and we can give it the qiskit purple from within airtable, then I think it's fine.

**Comment by y4izus** [comment]
Per offline conversation, we are gonna work on the airtable iframe option, so we can say this issue in completed. I close it. Great work @techtolentino !! Thank you!
